src/main/java/com/minecolonies/colony/buildings/BuildingGuardTower.java

```diff
@@ -0,0 +1,187 @@
+package com.minecolonies.colony.buildings;
+
+import com.minecolonies.client.gui.WindowHutWorkerPlaceholder;
+import com.minecolonies.colony.CitizenData;
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.ColonyView;
+import com.minecolonies.colony.jobs.AbstractJob;
+import com.minecolonies.colony.jobs.JobGuard;
+import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Building class of the guard tower.
+ */
+public class BuildingGuardTower extends AbstractBuildingWorker
+{
+    /**
+     * Name description of the guard hat.
+     */
+    private static final String GUARD_TOWER = "GuardTower";
+
+    /**
+     * Max level of the guard hut.
+     */
+    private static final int GUARD_HUT_MAX_LEVEL = 5;
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)